### DISCUSSION TITLE
JWT 토큰을 어디에 저장하는게 좋을까요

### DISCUSSION THREAD

**Discussion by juhee77** [MightyLions/Cold_Pitch#137] [Q&A] [answered]
지금은 헤더에 리프레시 토큰과 엑세스 토큰을 모두 넣고 있는데
찾아보다 보니 리프레시 토큰은 쿠키를 써서 쓰는게 나은것 같다는 말도 있더라구요
다른 분들은 보통 어떻게 하셨는지 궁금합니다.

**Comment by dlrjs2360** [comment]
redis가 가장 안전하다고 알고있는데 redis가 조금 과하다면 localstorage나 session은 xss공격에 취약하기 때문에 그나마 쿠키가 좋은 것 같습니당.

https://blogeon.tistory.com/entry/JWT%EC%9D%98-Refresh-Token%EA%B3%BC-Access-Token%EC%9D%80-%EC%96%B4%EB%94%94%EC%97%90-%EC%A0%80%EC%9E%A5%ED%95%B4%EC%95%BC-%ED%95%A0%EA%B9%8C

마침 여기 좋은 글이 하나 있습니다 하하

**Reply by juhee77** [reply]
혹시 헤더에 access , 쿠키에 refresh(http only) +RTR 는 어떠신가요 
글 정독하겠습니다

**Reply by dlrjs2360** [reply]
전 아주 좋다고 생각합니다. RTR까지 구현하신다면 아주 멋질 것 같습니다. 면접에서 할 말이 많겠군요.
글 정독은 사양합니다...자세히는 읽지 말아주세요

**Comment by juhee77** [comment]
근데 궁금한게 하나 더 있는데 엑세스 토큰이 만료된후에 이를 클라이언트가 알아차리고 서버에게 refreshtoken을 주면서 새로운 accesstoken,refreshtoken을 받는 방식이 맞는건가요? 아니면 서버측에서 엑세스 토큰이 만료된경우를 확인하고 쿠키에 있던 리프레시토큰을 이용해서 새로 발급을 제공하나요?? _spring security..._

**Reply by dlrjs2360** [reply, marked as answer]
두번째가 맞는 방식으로 알고있습니당. 요청마다 토큰이 헤더로 넘어오니까 그 때마다 필터에서 만료확인하고 발급까지 자동화하는 것으로 알고있어용

**Reply by juhee77** [reply]
이게 알람이 바로 가는군요.. 죄송합니다!! 감사해용!!!
